x-pack/platform/plugins/shared/fleet/server/services/security/uninstall_token_service/index.ts

```diff
@@ -309,10 +315,25 @@ export class UninstallTokenService implements UninstallTokenServiceInterface {
       ignoreMissing: true,
     });
 
-    return agentPolicies.reduce((dict, policy) => {
-      dict[policy.id] = policy.name;
+    const warnings: string[] = [];
+
+    const response = policyIds.reduce((dict, policyId) => {
+      const policy = agentPolicies.find((agentPolicy) => agentPolicy.id === policyId);
+
+      if (policy) {
+        dict[policyId] = policy.name;
+      } else {
+        warnings.push(`Policy id [${policyId}] was not found!`);
```

Comment:
So this also concerns me... There were no checks here to ensure that a policy was actually retrieved and it just retuned the result objects assuming that it did. 
I traced the call to this method here from:

```
getFullAgentPolicy() 
  --> UnstallTokenService.getHashedTokenForPolicyId() 
    --> UninstallTokenService.getHashedTokensForPolicyIds() 
      -->  UninstallTokenService.getDecryptedTokensForPolicyIds() 
        --> UninstallTokenService.getPolicyIdNameDictionary()
```

I'm not familiar with these signing and uninstall token logic, so its difficult for me to determine if it all is working as expected. My changes here will log a `.warn`ing, so if it is an issue, we will at least get some logging that this area of the code was unable to get the policy